src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -35,6 +35,11 @@ async function get_object(req, res) {
     if (part_number) {
         md_params.part_number = part_number;
     }
+
+    if (req.query.versionId?.length === 0) {
```

Comment:
The same check should also be in head object.
Best add it as a common function in s3_utils to avoid copy-pasta.
Also move it a bit up so it's with the other request query parsing cases.
```js
const version_id = s3_utils.parse_version_id(req.query.versionId);
```

Reply:
@guymguym why just head object and get object?
The function `s3_utils.parse_version_id(req);` is general and can be everywhere there is a parameter of `version_id`, for example: DeleteObject, DeleteObjectTagging, GetObjectAcl, PutObjectRetention, PutObjectTagging, RestoreObject (advance case ListObjectVersions - VersionIdMarker), etc.

We can search all `req.query.versionId` in the `src/endpoint/s3/ops`.

Reply:
But I don't think VersionIdMarker in listObjectVersions is the same.

Reply:
> But I don't think VersionIdMarker in listObjectVersions is the same.

@Neon-White I think It is something that we would need to check (from [AWS SDK V2 docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#listObjectVersions-property)):

> VersionIdMarker — (String)
> Specifies the object version you want to start listing from.

Reply:
NP, I think we can go ahead and replace all refs to req.query.versionId in s3 ops.

But for versionId in xml fields or as marker - I would first check how aws handles empty string there - I don't want to assume I know the answer.

Reply:
@Neon-White 

I tested on aws bucket -

1. delete-objects - does not return InvalidArgument at all with VersionId="", instead it just returns an array of errors with NoSuchVersion per requested version it cannot find - so we should undo the call to parse_version_id() in that op, and let our code continue with that empty string which will return the array of errors as it usually does -

```
❯ aws s3api delete-objects --bucket my-aws-bucket --delete '{"Objects":[ {"Key":"watman.png","VersionId":""}, {"Key":"README.md","VersionId":""} ]}'
{
    "Errors": [
        {
            "Key": "watman.png",
            "VersionId": "",
            "Code": "NoSuchVersion",
            "Message": "The specified version does not exist."
        },
        {
            "Key": "README.md",
            "VersionId": "",
            "Code": "NoSuchVersion",
            "Message": "The specified version does not exist."
        }
    ]
}
```

2. list-object-versions - VersionIdMarker is an input argument to this op and it does contain a versionId so it's not so different than other ops where versionId when passed as argument - I tested it with VersionIdMarker="" and it does reply with InvalidArgument with a similar yet more specific error message "A version-id marker cannot be empty". So this should actually be checked just like any other version id argument  -

```
❯ aws s3api list-object-versions --bucket my-aws-bucket --key-marker "watman.png" --version-id-marker ""

An error occurred (InvalidArgument) when calling the ListObjectVersions operation: A version-id marker cannot be empty.
```